### PR TITLE
Respect regex flags in validation and json schema

### PR DESCRIPTION
**Pull request by sydney-runkle**
Working on https://github.com/pydantic/pydantic/issues/9568

### CODE REVIEW

File: pydantic/json_schema.py [hunk 1]
```diff
@@ -660,6 +661,9 @@ def str_schema(self, schema: core_schema.StringSchema) -> JsonSchemaValue:
         """
         json_schema = {'type': 'string'}
         self.update_with_validations(json_schema, schema, self.ValidationsMapping.string)
+        if isinstance(json_schema.get('pattern'), Pattern):
+            # for now, we skip application of regex flags to the pattern
```
Comment (davidhewitt) [inline pydantic/json_schema.py:665]:
```suggestion
            # TODO: should we add regex flags to the pattern?
```

**Review by davidhewitt — COMMENTED** [review]
Overall looks fine to me, subject to core changes!

**Review by davidhewitt — APPROVED** [review]

**Comment by l-melon** [comment]
@sydney-runkle Hi, can I ask what `meetings.md` is used for?

**Comment by sydney-runkle** [comment]
@l-melon,

Great question, nothing!

**Pull request merged**